### PR TITLE
fix redirection in download tests

### PR DESCRIPTION
**Pull request by pmeier**
Before we never updated the initial `request` to incorporate the redirection

https://github.com/pytorch/vision/blob/9d278783e4883c6f2dd6646bd7002eae2e237efc/test/test_datasets_download.py#L56-L65

Furthermore, after #3560 has landed, we can now use `datasets.utils._get_redirect_url` instead of duplicating the functionality without `requests`.

### CODE REVIEW

File: test/test_datasets_download.py [hunk 1]
```diff
@@ -150,7 +162,7 @@ def assert_server_response_ok():
 
 
 def assert_url_is_accessible(url, timeout=5.0):
-    request = Request(url, headers={"method": "HEAD", "User-Agent": USER_AGENT})
+    request = Request(url, headers={"User-Agent": USER_AGENT}, method="HEAD")
```
Comment (fmassa) [inline test/test_datasets_download.py:165]: for my understanding: is this a cosmetic change?

**Review by fmassa — APPROVED** [review]
Thanks!

File: test/test_datasets_download.py (hunk 1 above)
Comment (pmeier) [reply test/test_datasets_download.py:165]: I didn't mean to include this. It was just for testing. But it shouldn't matter.

Comment (fmassa) [reply test/test_datasets_download.py:165]: Ok, if you want to revert this change in a follow-up PR I'm ok with it

**Pull request merged**